src/network/networkd-address.c

```diff
@@ -215,6 +215,32 @@ bool address_equal(Address *a1, Address *a2) {
         return address_compare_func(a1, a2) == 0;
 }
 
+static int address_copy(Address *dest, Address *src) {
+        _cleanup_free_ char *label = NULL;
+
+        assert(dest);
+        assert(src);
+
+        if (src->label) {
+                label = strdup(src->label);
+                if (!label)
+                        return -ENOMEM;
+        }
+
+        dest->family = src->family;
+        dest->prefixlen = src->prefixlen;
+        dest->scope = src->scope;
+        dest->flags = src->flags;
+        free_and_replace(dest->label, label);
```

Comment:
Maybe use `free_and_strdup` first, and then assign the other fields if that works? (`free_and_strdup` doesn't modify the output on failure, so it'd give the same atomicity.)

Reply:
Done.